source/gitlab/gitlab.go

```diff
@@ -52,29 +52,26 @@ func (g *Gitlab) Open(url string) (source.Driver, error) {
 		return nil, ErrNoUserInfo
 	}
 
+	if u.Host == "" {
+		return nil, ErrInvalidHost
+	}
+
 	password, ok := u.User.Password()
 	if !ok {
 		return nil, ErrNoAccessToken
 	}
 
+	client, err := gitlab.NewClient(password, gitlab.WithBaseURL(u.String()))
```

Comment:
The password should be stripped from the URI string.

Reply:
Ahh good catch. I updated it to use `Redacted()`.